File: sdk/java/examples/StartProfile.java
// Multilogin X — Start browser profile (OkHttp)
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class StartProfile {
    public static void main(String[] args) throws Exception {
        String host = env("MLX_LAUNCHER_HOST", "launcher.mlx.yt");
        String port = env("MLX_LAUNCHER_PORT", "45001");
        String folderId = require("MLX_FOLDER_ID");
        String profileId = require("MLX_PROFILE_ID");
        String automation = env("MLX_AUTOMATION_TYPE", "puppeteer");
        String headless = env("MLX_HEADLESS", "false");

        String url = String.format(
            "https://%s:%s/api/v2/profile/f/%s/p/%s/start?automation_type=%s&headless_mode=%s",
            host, port, folderId, profileId, automation, headless
        );

        Request request = new Request.Builder()
            .url(url)
            .get()
            .addHeader("Accept", "application/json")
            .build();

        try (Response response = new OkHttpClient().newCall(request).execute()) {
            if (!response.isSuccessful()) {
                throw new RuntimeException("HTTP " + response.code());
            }
            System.out.println(response.body().string());
        }
    }

    private static String env(String key, String fallback) {
        String v = System.getenv(key);
        return v != null && !v.isBlank() ? v : fallback;
    }

    private static String require(String key) {
        String v = System.getenv(key);
        if (v == null || v.isBlank()) {
            throw new IllegalStateException("Set " + key);
        }
        return v;
    }
}

File: sdk/csharp/examples/StartProfile.cs
// Multilogin X — Start browser profile (HttpClient)
using System;
using System.Net.Http;
using System.Threading.Tasks;

var host = Environment.GetEnvironmentVariable("MLX_LAUNCHER_HOST") ?? "launcher.mlx.yt";
var port = Environment.GetEnvironmentVariable("MLX_LAUNCHER_PORT") ?? "45001";
var folderId = Environment.GetEnvironmentVariable("MLX_FOLDER_ID")
    ?? throw new InvalidOperationException("Set MLX_FOLDER_ID");
var profileId = Environment.GetEnvironmentVariable("MLX_PROFILE_ID")
    ?? throw new InvalidOperationException("Set MLX_PROFILE_ID");
var automation = Environment.GetEnvironmentVariable("MLX_AUTOMATION_TYPE") ?? "puppeteer";
var headless = Environment.GetEnvironmentVariable("MLX_HEADLESS") ?? "false";

var url =
    $"https://{host}:{port}/api/v2/profile/f/{folderId}/p/{profileId}/start" +
    $"?automation_type={automation}&headless_mode={headless}";

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Get, url);
request.Headers.Add("Accept", "application/json");

var response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
Console.WriteLine(await response.Content.ReadAsStringAsync());
